src/views/Rules/ListRules/index.jsx

```diff
@@ -249,9 +297,35 @@ function ListRules(props) {
       throw error;
     }
 
-    setRulesWithScheduledChanges(
-      rulesWithScheduledChanges.filter(i => i.rule_id !== dialogRule.rule_id)
-    );
+    if (Object.keys(dialogRule.requiredSignoffs).length > 0) {
+      // A change was scheduled, we need to update the card
+      // to reflect that.
+      try {
+        const sc = (await getScheduledChange(dialogRule.rule_id)).data
+          .scheduled_changes[0];
+
+        setRulesWithScheduledChanges(
+          rulesWithScheduledChanges.map(i => {
+            if (i.rule_id !== sc.rule_id) {
+              return i;
+            }
+
+            i.scheduledChange = sc;
+
+            return i;
+          })
+        );
+      } catch (e) {
+        throw new Error(
+          'Scheduled change created but page failed to properly update page. Please refresh.'
+        );
+      }
+    } else {
+      // The rule was directly deleted, just remove it.
+      setRulesWithScheduledChanges(
+        rulesWithScheduledChanges.filter(i => i.rule_id !== dialogRule.rule_id)
```

Comment:
What does `i` stand for? If it's a rule, then a nit would be to probably use `rule` or `r` instead for readability. When I saw `i` I thought it meant index :)

Reply:
Item? :smile: I agree that `r` is better.